File: mobile-missao/src/interfaces/alertaCritico.ts
import { NivelAlerta } from "../types/nivelAlerta";
import { StatusAlerta } from "../types/statusAlerta";

export interface AlertaCritico {
  id: number;
  titulo: string;
  descricao: string;
  nivel: NivelAlerta;
  status: StatusAlerta;
  dataHora: string;
}

File: backend-missao/src/main/java/com/fiap/ec/backend_missao/model/AlertaCritico.java
package com.fiap.ec.backend_missao.model;

import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "alertas_criticos")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class AlertaCritico {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String titulo;
    private String descricao;
    private String nivel;
    private String status;
    private LocalDateTime dataHora;
}